pkg/cli/admin/mustgather/mustgather.go

```diff
@@ -579,36 +583,41 @@ func (o *MustGatherOptions) Run() error {
 	}
 
 	// print at both the beginning and at the end.  This information is important enough to be in both spots.
-	o.PrintBasicClusterState(context.TODO())
+	o.PrintBasicClusterState(ctx)
 	defer func() {
+		// Shortcircuit this in case the context is already cancelled.
+		if ctx.Err() != nil {
+			return
+		}
 		fmt.Fprintf(o.RawOut, "\n\n")
 		fmt.Fprintf(o.RawOut, "Reprinting Cluster State:\n")
-		o.PrintBasicClusterState(context.TODO())
+		o.PrintBasicClusterState(ctx)
 	}()
 
 	// Ensure resource cleanup unless instructed otherwise ...
-	var cleanupNamespace func()
+	var cleanupNamespace func(context.Context)
```

Comment:
I think we shouldn't wire any context to prevent accidental incorrect wirings to cleanupNamespace. 

cleanupNamespace always should have its own dedicated context and you explicitly pass a separate one in line:602 already.